test/jasmine/tests/transform_aggregate_test.js

```diff
@@ -0,0 +1,225 @@
+        expect(traceOut.text).toEqual(['a', 'b']);
+        expect(traceOut.hovertext).toEqual(['c', 'b']);
+        expect(traceOut.marker.line.width).toEqual([4, 1]);
+    });
+
+    it('allows date and category sums, and category avg, with weird output', function() {
+        // this test is more of an FYI than anything else - it doesn't break but
+        // these results are usually meaningless.
+
+        Plotly.newPlot(gd, [{
+            x: ['2001-01-01', '2001-01-02', '2001-01-03', '2001-01-04'],
+            y: ['a', 'b', 'b', 'c'],
+            text: ['a', 'b', 'a', 'c'],
+            transforms: [{
+                type: 'aggregate',
+                groups: [1, 1, 2, 2],
+                aggregations: [
+                    {target: 'x', func: 'sum'},
+                    {target: 'y', func: 'sum'},
+                    {target: 'text', func: 'avg'}
+                ]
+            }]
+        }]);
+
+        var traceOut = gd._fullData[0];
+
+        // date sums: 1970-01-01 is "zero", there are shifts due to # of leap years
+        // without that shift these would be 2032-01-02 and 2032-01-06
+        expect(traceOut.x).toEqual(['2032-01-03', '2032-01-07']);
+        // category sums: can go off the end of the category array -> gives undefined
+        expect(traceOut.y).toEqual(['b', undefined]);
+        // category average: can result in fractional categories -> rounds (0.5 rounds to 1)
+        expect(traceOut.text).toEqual(['b', 'b']);
+    });
+
+    it('can aggregate on an existing data array', function() {
+        Plotly.newPlot(gd, [{
+            x: [1, 2, 3, 4, 5],
+            y: [2, 4, 6, 8, 10],
+            marker: {size: [10, 10, 20, 20, 10]},
+            transforms: [{
+                type: 'aggregate',
+                groups: 'marker.size',
+                aggregations: [
+                    {target: 'x', func: 'sum'},
+                    {target: 'y', func: 'avg'}
+                ]
+            }]
+        }]);
+
+        var traceOut = gd._fullData[0];
+
+        expect(traceOut.x).toEqual([8, 7]);
+        expect(traceOut.y).toBeCloseToArray([16 / 3, 7], 5);
+        expect(traceOut.marker.size).toEqual([10, 20]);
+    });
+
+    it('handles median, mode, rms, & stddev for numeric data', function() {
+        // again, nothing is going to barf with non-numeric data, but sometimes it
+        // won't make much sense.
+
+        Plotly.newPlot(gd, [{
+            x: [1, 1, 2, 2, 1],
+            y: [1, 2, 3, 4, 5],
+            marker: {
+                size: [1, 2, 3, 4, 5],
+                line: {width: [1, 1, 2, 2, 1]}
+            },
+            transforms: [{
+                type: 'aggregate',
+                groups: [1, 2, 1, 1, 1],
+                aggregations: [
+                    {target: 'x', func: 'mode'},
+                    {target: 'y', func: 'median'},
+                    {target: 'marker.size', func: 'rms'},
+                    {target: 'marker.line.width', func: 'stddev'}
```

Comment:
This should be fine. Yes, we'll have to describe it in detail in `description`, and manage it dynamically in `supplyDefaults`, but that seems to me better than making different names for each `func` that gets varying `mode`s.